### PR TITLE
fix: Lazy load mpc core kit

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Part of #2452 

Lazy loads the mpc core kit to reduce the initial js bundle size

## How this PR fixes it

- Converts the `useInitMPC` and `useInitSocialWallet` hooks into functions
- Uses `next/dynamic` to import `SocialSigner`
- Changes the onboard module to lazy import dependencies

## How to test it

1. Open the safe
2. Login with social
3. Observe being logged in
4. Reload the page
5. Observe being logged in with social again
6. Log in with Metamask or any other wallet
7. Observe it still working

### CODE REVIEW

File: src/components/common/ConnectWallet/WalletDetails.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { Box, Divider, SvgIcon, Typography } from '@mui/material'
+import dynamic from 'next/dynamic'
 import type { ReactElement } from 'react'
 
 import LockIcon from '@/public/images/common/lock.svg'
-import SocialSigner from '@/components/common/SocialSigner'
+
+const SocialSigner = dynamic(() => import('@/components/common/SocialSigner'))
```
[the remark this answers is not in the document]
Comment (usame-algan) [reply src/components/common/ConnectWallet/WalletDetails.tsx:7]: In the docs it says that `next/dynamic` is a composite of `React.Lazy` and `Suspense` so I would assume it doesn't need another Suspense: https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#nextdynamic

File: src/hooks/wallets/mpc/useMPC.ts [hunk 2]
```diff
@@ -1,88 +1,82 @@
...
 const { getStore, setStore, useStore } = new ExternalStore<Web3AuthMPCCoreKit>()
 
-export const useInitMPC = () => {
-  const chain = useCurrentChain()
-  const onboard = useOnboard()
-  useInitSocialWallet()
+export const initMPC = async (chain: ChainInfo, onboard: OnboardAPI) => {
+  const chainConfig = {
+    chainId: `0x${Number(chain.chainId).toString(16)}`,
+    chainNamespace: 'eip155',
```
Comment (katspaugh) [inline src/hooks/wallets/mpc/useMPC.ts:17]: Please defne this as a const.

File: src/hooks/wallets/mpc/useRehydrateSocialWallet.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          const address = wallets[0].accounts[0]?.address
+          if (address) {
+            const signerAddress = checksumAddress(address)
+            if (addressBook[signerAddress] === undefined) {
+              const email = userInfo.email
+              dispatch(upsertAddressBookEntry({ address: signerAddress, chainId: currentChainId, name: email }))
+            }
+          }
+        }
+      }
```
Comment (katspaugh) [inline src/hooks/wallets/mpc/useRehydrateSocialWallet.ts:50]: Please extract this into a function (can be within the hook).

File: src/hooks/wallets/mpc/useMPC.ts (hunk 2 above)
Comment (schmanu) [reply src/hooks/wallets/mpc/useMPC.ts:17]: We could probably revert this to the constant we were using.

File: src/hooks/wallets/mpc/useRehydrateSocialWallet.ts (hunk 3 above)
Comment (usame-algan) [reply src/hooks/wallets/mpc/useRehydrateSocialWallet.ts]: done

**Review by katspaugh — APPROVED** [review]

**Review by schmanu — APPROVED** [review]
I played around with it for a bit. 
It works well. 
One small issue I encountered is that the login with google button is not displayed / on a fresh page load and also no placeholder.

**Comment by usame-algan** [comment]
I've also updated the bundle-analyzer action to use the sticky-comment workflow which is set to be released with https://github.com/hashicorp/nextjs-bundle-analysis/pull/54

**Pull request merged**